code/go/0chain.net/miner/m_handler_integration_tests.go

```diff
@@ -87,6 +95,22 @@ func NotarizationReceiptHandler(ctx context.Context, entity datastore.Entity) (i
 		}()
 	}
 
+	state := crpc.Client().State()
+
+	if state.RoundHasFinalized != nil && state.RoundHasFinalized.Round == int(not.Round) && utils.IsSpamReceiver(state, not.Round) {
+		m := GetMinerChain()
+		mr := m.getOrCreateRound(ctx, (int64)(state.RoundHasFinalized.Round+1))
```

Comment:
so we get the next to finalized round and wait for VRF there, but it's not correct, since this round is guaranteed to be notarized and it is finished, we need the next round to it, so if r0 is finalized the current round is r0+2 or r0+3. Why we can't take simply current round and wait for VRFs there?

Reply:
Here we are putting on hold the handling of the notarization of the current round to prevent the current round from being finalized. Only after receiving the spammed VRF of the next round, we will proceed to the handling.

I guess the idea was to lock the notarization. @dabasov what do you suggest to do?